File: verify_hq.py
import time
import json
from playwright.sync_api import sync_playwright, TimeoutError as PlaywrightTimeoutError

def setup_and_verify(page):
    print("Navigating to dev server...")
    page.goto('http://localhost:5173/')

    print("Checking for SKIP INTRO...")
    try:
        page.get_by_text("SKIP INTRO").click(timeout=3000)
        print("Clicked SKIP INTRO.")
    except PlaywrightTimeoutError:
        print("No SKIP INTRO button found.")

    print("Checking for SKIP ALL (tutorial)...")
    try:
        page.get_by_text("SKIP ALL").click(timeout=3000)
        print("Clicked SKIP ALL.")
    except PlaywrightTimeoutError:
        print("No SKIP ALL button found.")

    print("Waiting for BAND HQ button...")
    try:
        page.get_by_role("button", name="BAND HQ").click(timeout=5000)
        print("Clicked BAND HQ.")
    except PlaywrightTimeoutError:
        print("Could not find BAND HQ button. Taking debug screenshot...")
        page.screenshot(path='/home/jules/verification/hq_debug.png')
        raise

    time.sleep(1)

    print("Looking for UPGRADES tab...")
    try:
        page.locator('button:has-text("UPGRADES")').click(timeout=5000)
        print("Clicked UPGRADES tab.")
    except PlaywrightTimeoutError:
        print("Could not find UPGRADES tab. Taking debug screenshot...")
        page.screenshot(path='/home/jules/verification/hq_debug_tabs.png')
        raise

    time.sleep(1)

    print("Looking for HQ tab...")
    try:
        page.get_by_role("button", name="HQ").click(timeout=3000)
        print("Clicked HQ tab.")
    except PlaywrightTimeoutError:
        print("No inner HQ tab found or already on it.")

    time.sleep(1)

    print("Taking final screenshot...")
    page.screenshot(path='/home/jules/verification/pr_manager_contract.png')
    print("Verification complete.")

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)

    context = browser.new_context()
    page = context.new_page()

    page.goto('http://localhost:5173/')

    injected_state = {
      "version": "1.0",
      "player": {
        "money": 10000,
        "fame": 1000,
        "hqLevel": 1,
        "inventory": [],
        "activeEffects": []
      },
      "stats": {
        "currentScene": "OVERWORLD"
      }
    }

    state_str = json.dumps(injected_state)
    page.evaluate(f"localStorage.setItem('neurotoxic_v3_save', JSON.stringify({state_str}))")

    page.reload()

    try:
        setup_and_verify(page)
    except Exception as e:
        print(f"Error: {e}")
    finally:
        browser.close()

File: verify_hq.js
import { chromium } from 'playwright';

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();

  // Navigate to local dev server
  await page.goto('http://localhost:5173/');

  // Skip intro
  console.log('Skipping intro...');
  await page.waitForSelector('text="SKIP INTRO"');
  await page.click('text="SKIP INTRO"');

  // Skip tutorial if present
  try {
    const skipAll = await page.waitForSelector('text="SKIP ALL"', { timeout: 2000 });
    if (skipAll) {
      console.log('Skipping tutorial...');
      await skipAll.click();
    }
  } catch (e) {
    console.log('No tutorial skip button found.');
  }

  // Open Band HQ
  console.log('Opening Band HQ...');
  await page.waitForSelector('button:has-text("BAND HQ")');
  await page.click('button:has-text("BAND HQ")');

  // Wait for the modal to be visible
  await page.waitForSelector('text="BAND HQ"');
  await page.waitForTimeout(1000); // Let UI settle

  // Click on UPGRADES tab
  console.log('Switching to UPGRADES tab...');
  await page.locator('button:has-text("UPGRADES")').click();

  await page.waitForTimeout(1000); // Let UPGRADES load

  // Click on HQ tab if there are sub-tabs (like VAN and HQ)
  console.log('Switching to HQ tab...');
  try {
     // sometimes the subtabs are 'HQ' and 'VAN'
     await page.locator('button:has-text("HQ")').click();
     await page.waitForTimeout(1000);
  } catch (e) {
     console.log('No sub-HQ tab found, proceeding...');
  }

  console.log('Taking screenshot...');
  await page.screenshot({ path: '/home/jules/verification/hq_upgrades.png' });

  console.log('Verification Complete.');
  await browser.close();
})();
